packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,129 @@
+      OUTPUT_CHANNEL
+    );
+    this.sourceFsPath = sourceFsPath;
+    this.responseText = responseText;
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: vscode.Progress<{ message?: string | undefined; increment?: number | undefined; }>,
+    token?: vscode.CancellationToken
+    ): Promise<boolean> {
+      if (this.sourceFsPath) {
+        if (this.responseText) {
+          renameComponent(this.sourceFsPath, this.responseText);
+          return true;
+        }
+      }
+      return false;
+  }
+}
+
+export async function forceRenameLightningComponent(sourceUri: vscode.Uri) {
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(REAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  if (items) {
+    for (const item of items) {
+      // only rename the file that has same name with component
+      if (isNameMatch(item, componentName, componentPath)) {
+        const newItem = item.replace(componentName, newName);
+        fs.renameSync(
+          path.join(componentPath, item),
+          path.join(componentPath, newItem));
+      }
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath);
+  notificationService.showWarningMessage(nls.localize(RENAME_COMP_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    notificationService.showErrorMessage(nls.localize(REANME_INPUT_DUP_ERROR));
+    throw new Error(format(nls.localize(REANME_INPUT_DUP_ERROR)));
+  }
+}
+
+function isDuplicate(componentPath: string, newName: string): boolean {
+  const isLwc = isLwcComp(componentPath);
+  const lwcPath = isLwc ? path.dirname(componentPath) : path.join(path.dirname(path.dirname(componentPath)), 'lwc');
+  const auraPath = isLwc ? path.join(path.dirname(path.dirname(componentPath)), 'aura') : path.dirname(componentPath);
+  if (fs.existsSync(path.join(lwcPath, newName)) || fs.existsSync(path.join(auraPath, newName))) {
+    return true;
+  }
+  return false;
```

Comment:
There are are lot "if LWC else Aura" taking place here.  I think the LWC code should be moved into its own function, and the Aura code moved into its own function.  So something like this:
```
function isDuplicate(componentPath: string, newName: string): boolean {
  if (isLwcComp(componentPath)) {
    return isDuplicateLwcComponent((componentPath, newName);
  } else {
    return isDuplicateAuraComponent((componentPath, newName);
  }
```

Reply:
@jeffb-sfdc We need to check duplication in bot LWC and Aura path instead of either of them. But, I revised my code to use only one "if lwc else" logic here to add more readability.

Reply:
With the updated code, there looks to be a logic flip -the "isLwcComponent is true" condition has:
`auraPath = path.join(path.dirname(componentPathDirName), 'aura');`

...shouldn't it be lwc instead of Aura?

Reply:
nm, I think I understand now... when renaming, we need to check for the name in `/lwc` as well as `/aura`.